bpf-loader-rs/bpf-loader-lib/src/skeleton/preload/attach.rs

```diff
@@ -30,10 +32,36 @@ impl Drop for AttachLink {
                     error!("Failed to destroy tc hook: {}", err);
                 }
             }
+            AttachLink::XDPAttach(ifindex, flags, opts) => {
+                let err = unsafe { bpf_xdp_detach(*ifindex, *flags, &**opts) };
+                if err != 0 {
+                    error!("Failed to detach xdp hook");
+                }
+            }
         }
     }
 }
 
+pub(crate) fn attach_xdp(program: &Program, meta: &ProgMeta) -> Result<AttachLink> {
+    let xdp_extra_meta = serde_json::from_value::<XDPProgExtraMeta>(meta.others.clone())
+        .with_context(|| anyhow!("Failed to deserialize xdp extra meta"))?;
+
+    let ifindex = xdp_extra_meta.ifindex;
+    let flags = xdp_extra_meta.flags;
+    let prog_fd = program.fd();
+
+    let mut xdp_attach_opts = Box::new(unsafe { std::mem::zeroed::<bpf_xdp_attach_opts>() });
+    xdp_attach_opts.sz = std::mem::size_of::<bpf_xdp_attach_opts>() as _;
+    xdp_attach_opts.old_prog_fd = xdp_extra_meta.xdpopts.old_prog_fd;
+
+    let err = unsafe { bpf_xdp_attach(ifindex, prog_fd, flags, &*xdp_attach_opts) };
```

Comment:
Unsafe blocks should be described with a "//SAFETY:" comment to provide a brief description on what this unsafe call does and why this unsafe call is "safe"

Reply:
See other unsafe calls for reference